File: keepr/Models/Keep.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace keepr.Models;

public class Keep : RepoItem<int>
{
    public string Name { get; set; }
    public string Description { get; set; }
    public string Img { get; set; }
    public string CreatorId { get; set; }
    public int Views { get; set; }
    public Profile Creator { get; set; }
}

public class VaultedKeep : Keep
{
    public int vaultKeepId { get; set; }
}

File: keepr.client/src/models/Keep.js
export class Keep {
  constructor(data) {
    this.id = data.id
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.creatorId = data.creatorId
    this.views = data.views
    this.kept = data.kept
    this.creator = data.creator
    this.vaultKeepId = data.vaultKeepId
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
  }
}